bsip-0074.md

```diff
@@ -0,0 +1,36 @@
+```
+BSIP: 0074
+Title: Margin Call Fee Ratio
+Authors:
+  Jerry Liu bitcrab@qq.com
+Status: Draft
+Type: Protocol
+Created: 2019-09-07
+Discussion: https://github.com/bitshares/bsips/issues/164
+Worker: TBD
+```
+
+# Abstract
+This BSIP provide a solution to charge fee from margin call trading.
+
+# Motivation
+Bitshares community becomes more and more aware that BTS need a powerful economy model, one simple selection is to charge the tradings and buy back BTS using the accumulated fees, as did in many exchange platform tokens, the continuous income can help to provide smartcoin liquidity and sustain BTS price. Margin call tradings are potential to contribute a big part of the income to the system, however up to now the margin call trading is not charged, we need to find a suitable way to get this part of system income.
+
+# Rationale
+Margin call fee can be seen as to pay the cost for smartcoin supply and stabilization, it is irrelevant to market fee sharing.
+
+To ensure the debt positions can always be closed with suffcient smartcoin, it is more feasible to cut off part of the traded collaterals as fee, instead of smartcoin.
+
+As the margin call fee ratio may be obvious higher than market fee, the margin call orders should be placed at the real price without margin call fee to avoid misleading the buyer.
+
+# Specification
+Add one new parameter Margin Call Fee Ratio(MCFR) for each smartcoin, which is controlled by the smartcoin owner.
+
+`Margin call order price = settlement price/(MSSR-MCFR)`.
+
+Here settlement price is a new introduced parameter which is defined in BSIP71, settlement price = feed price when there is no bad debt.
+
+When a margin call trading happens, the buyer sells smartcoin with quantity X and get collaterals in quantity `X*(MSSR-MCFR)/settlement price`, the margin call order owner sells collaterals in quantity `X*MSSR/settlement price` and get smartcoin in quantity X, the delta in paid and received collaterals in `quantity X*MCFR/settlement price` will be paid to system as margin call fee.
```

Comment:
Please explain what "paid to system" means, i. e. where the fee goes.
Note that the collateral asset is not necessarily BTS, i. e. it can not generally go into the reserve pool nor into the fee pool of the asset.

Reply:
oh, yes, it should be "paid to the owner of the smartcoin".

Reply:
Fees are usually paid into a vesting balance from which the owner can later claim them. Would that be ok here too?
(The reason is that it simplifies account history. Without the vesting balance you'd have an entry for each individual fee payment, which can be a lot.)

Reply:
that's ok.